test/functional/provider/admin/messages/inbox_controller_test.rb

```diff
@@ -48,4 +48,24 @@ def test_index_system_message
     assert_select 'title', "Inbox - Index | Red Hat 3scale API Management"
     assert_select '#export-to-csv', false, 'Export all Messages'
   end
+
+  test 'Prevent sending reply without subject' do
+    login_as(@admin)
+    reply = @message.reply
+    reply.update_attributes(body: 'Reply Body', subject: nil)
```

Comment:
`@message.reply` [creates a new Message instance](https://github.com/3scale/porta/blob/master/app/models/message_recipient.rb#L75-L79) with subject and body based on the `@message`. By using `update_attributes` on it, if there are no errors, the message will be stored on the DB, which doesn't look like something we want to do, as the code that was updated on `Provider::Admin::Messages::InboxController` calls `Message#reply` as well, creating a new Message instance as well.

Reply:
Ummm you are bit wrong here because we have added validation on model for subject can't be nil so here as you said that @message.reply creates a new instance then when update attribute is called on same and if any validation fail it will rollback it because reply is not saved yet  and will not save it.

Here attaching screenshot for you to understand it better

![image](https://user-images.githubusercontent.com/84672731/199464169-2dd131d5-6ac5-4433-af39-48d98621f02c.png)

Reply:
Yes. That's what I meant with "... By using update_attributes on it, **if there are no errors**, the message will be stored on the DB..."

The point here is that this file contains the tests for `Provider::Admin::Messages::InboxController` but we're not really testing the controller actions.
For example, on the test `'renders index page with export option for admins'`, we call `get :index` to call the controller action.
As per https://github.com/3scale/porta/pull/3094/files#r1009591764, we should call `post :reply` for the new tests and assert they behave as expected.

Also, if it makes sense, part of the code you added in here could be used as unit tests 🎉

Reply:
yes I got your point and test are updated